GSA/digitalgov.gov
# Fix glossary header bug

## Summary
Guide pages that display a `summary-box` component also display the **Glossary** `h2` when it should not be visible.
`glossary.html` is unnecessarily rendered on guide pages that don't use a glossary.


See [guides/dap/common-questions-about-dap](https://digital.gov/guides/dap/common-questions-about-dap/)
<img width="1141" alt="Screen Shot 2024-03-01 at 2 08 39 PM" src="https://github.com/GSA/digitalgov.gov/assets/104778659/f6c57a12-5b70-485c-9743-13230b488e4d">


### Preview
[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-fix-guide-glossary-bug/guides/dap/common-questions-about-dap/)

### Solution
1. Add a conditional check to display `glossary.html` when `.Parent.Params.glossary` is true.
2. If the `summary-box` component is used on a page with a `glossary` component enabled, then don't display the **Glossary** `h2`.
3. Hide the **Glossary** `h2` in `summary-box.js`

### How To Test

1. Visit [guide page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-fix-guide-glossary-bug/guides/dap/common-questions-about-dap/) and should not see **glossary** header in `summary box` when glossary **is not** enabled.
2. Visit [guide page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-fix-guide-glossary-bug/guides/hcd/discovery-concepts/present/) and should not see **glossary** header in `summary box` when glossary **is** enabled.
	- set `summary_box: true` on `content/guides/hcd/discovery-concepts/present.md` on your local machine
2. Visit [guide landing page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-fix-guide-glossary-bug/guides/hcd/introduction/) and should not see **glossary** header in `summary box` when glossary **is** enabled.
	- set `summary_box: true` on `content/guides/hcd/introduction/_index.md` on your local machine

### What To Test

- [ ] `glossary.html` is not rendered when `glossary` field is not set
	- `dg-glossary__container` is not present in markup
- [ ] `summary-box` component does not display **Glossary** `h2` at all (even when set).



---

### Dev Checklist
